src/controlobjectscript.h

```diff
@@ -0,0 +1,38 @@
+#ifndef CONTROLOBJECTSCRIPT_H
+#define CONTROLOBJECTSCRIPT_H
+
+#include "controllers/controllerengine.h"
+
+#include "controlobjectslave.h"
+
+// this is used for communicate with controller scripts
+class ControlObjectScript : public ControlObjectSlave {
+    Q_OBJECT
+  public:
+    ControlObjectScript(const ConfigKey& key, QObject* pParent = NULL);
```

Comment:
Constructor with a single parameter -> "explicit"
The optional parameters do not count as parameters when applying this rule ;)